ynr/apps/elections/templates/elections/ballot_view.html

```diff
@@ -91,6 +94,17 @@ <h1>Candidates for {{ ballot.post.label }}</h1>
               </a>
             </td>
             <td>{{ candidate.party.name }}</td>
+            
+            {% if ballot.is_welsh_run and ballot.candidates_locked %}
+              {% for membership in candidate.memberships %}
+                {% for previous_party in membership.previous_party_affiliations %}
```

Comment:
Try adding `.all` on the end of this line.

Also, in order to add some `previous_party_affiliations` relationships at the moment this is only really possible via a shell. There are a few ways to add ManyToMany relationships so its [worth reading through the docs](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/). But here's an example of how I have been doing this via a shell:
```
# get a candidacy for a ballot in wales
m = Membership.objects.filter(
     ballot__ballot_paper_id__startswith="local.",
     ballot__tags__NUTS1__value="Wales",
).first()
# add relationship to "Independent" party object
m.previous_party_affiliations.add(Party.objects.get(ec_id="ynmp-party:2"))
# then check which person/ballot it is and visit in browser to test
m.person.get_absolute_url()
'/person/22622/brian-makusha'
m.ballot.get_absolute_url()
'/elections/local.powys.llandrindod-eastllandrindod-west.2017-05-04/'

```